File: python/src/sorts/sorts.py
def bubble(arr):
    print(f"Bubble sort started")
    n = len(arr)
    swapped = False
    for i in range(n-1):
        for j in range(0, n-i-1):
            if arr[j] > arr[j + 1]:
                swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
        if not swapped:
            return

def quick(arr, low, high):
    print(f"Quick sort with {low}, {high}")
    if low < high:
        pi = partition(arr, low, high)
        quick(arr, low, pi - 1)
        quick(arr, pi + 1, high)

def partition(arr, low, high):
    pivot = arr[high]
    i = low - 1
    for j in range(low, high):
        if arr[j] <= pivot:
            i = i + 1
            (arr[i], arr[j]) = (arr[j], arr[i])

    (arr[i + 1], arr[high]) = (arr[high], arr[i + 1])

    return i + 1

import time
def sleep(arr):
    for i in range(1, len(arr) + 1):
        print(f"Sleep {i} time(s)")
        time.sleep(1)

File: go/workflows/sorts.go
package workflows

func bubble(arr[] int)[]int {
   for i:=0; i< len(arr)-1; i++ {
      for j:=0; j < len(arr)-i-1; j++ {
         if (arr[j] > arr[j+1]) {
            arr[j], arr[j+1] = arr[j+1], arr[j]
         }
      }
   }
   return arr
}

func quick(arr []int, low, high int) []int {
	if low < high {
		var p int
		arr, p = partition(arr, low, high)
		arr = quick(arr, low, p-1)
		arr = quick(arr, p+1, high)
	}
	return arr
}

func partition(arr []int, low, high int) ([]int, int) {
	pivot := arr[high]
	i := low
	for j := low; j < high; j++ {
		if arr[j] < pivot {
			arr[i], arr[j] = arr[j], arr[i]
			i++
		}
	}
	arr[i], arr[high] = arr[high], arr[i]
	return arr, i
}
